Remove IP Specification From Compose Ports

HassanAbouelela opened the pull request:
This PR originally disabled postgres' durable optimizations for a significant speedup in testing, but more experimentation proved that it was the port changes that had the most effect. There is more info below, though the reason this works remains a bit mysterious.

MarkKoz commented on docker-compose.yml:
```diff
@@ -42,12 +53,15 @@ services:
       postgres:
         condition: service_healthy
+      test-postgres:
+        condition: service_started
```
Does this mean it's going to run two instances of the db?

HassanAbouelela replied:
Yes, it does. There's the option of not adding this to the depends-on and letting people manually invoke it if we don't want that.

MarkKoz replied:
Well, Docker Compose is intended to only be used for development purposes, right? So why don't we just always disable durability? I don't think anyone would cry over development data being lost in the rare case of power loss or a crash.

HassanAbouelela replied:
That's a fair point, I don't see a good reason not to.

MarkKoz commented on .github/workflows/lint-test.yaml:
```diff
@@ -97,13 +97,13 @@ jobs:
-          DATABASE_URL: postgres://pysite:pysite@localhost:7777/pysite
-          METRICITY_DB_URL: postgres://pysite:pysite@localhost:7777/metricity
+          DATABASE_URL: postgres://pysite:pysite@localhost:7778/pysite
+          TEST_DB_URL: postgres://pysite:pysite@localhost:7778/pysite
```
What's the point of separate URLs if they're identical anyway? Do we _need_ the flexibility to make them different if we want?

HassanAbouelela replied:
They're both needed as one is used in the migrations, and one is used in the tests. Making them one environment variable would be annoying as locally and in the compose they aren't going to refer to the same thing.

MarkKoz replied:
Oh sorry, I see that they are different by default in docker-compose.yml. I confused this file with docker-compose.yml.

HassanAbouelela committed:
Remove IP From Compose Ports

Specifying the port IP in compose seems to significantly slow down
networking with postgres. This effect is epically noticeable in tests,
which are 5-10x faster with the change.

Signed-off-by: Hassan Abouelela <hassan@hassanamr.com>

MarkKoz commented on docker-compose.yml:
```diff
@@ -8,12 +8,12 @@
 # unsuitable for production.
 
-version: "3.6"
+version: "3.8"
```
For future reference, there isn't really a point in changing this as its deprecated

https://github.com/compose-spec/compose-spec/blob/master/spec.md#version-top-level-element

> A Compose implementation SHOULD NOT use this version to select an exact schema to validate the Compose file, but prefer the most recent schema at the time it has been designed.

Maybe there are still some tools that are not respecting this advice?

HassanAbouelela replied:
My compose does use the behavior you described above, so I'd be happy to remove it. The reason I bumped it is because our current compose is using features that weren't even introduced in 3.6, and I wanted to fix the linting issues.

MarkKoz replied:
Does `docker-compose` still rely on it (notice the dash)? Maybe wait until that's completely phased out? I don't really know. It doesn't hurt to leave it for now I guess.

MarkKoz commented on pyproject.toml:
```diff
@@ -51,7 +51,7 @@ start = "python manage.py run --debug"
 django_shell = "python manage.py shell"
 test = "coverage run manage.py test"
-coverage = "coverage run manage.py test --no-input; coverage report -m"
+coverage = "coverage run manage.py test --no-input && coverage report -m"
```
Would be nice if this ran migrations too.

HassanAbouelela replied:
That sounds fine. Would you prefer the initial approach of putting it in the manage.py, or should we do it here?

MarkKoz replied:
Doing it in `manage.py` would be better so it still works without reliance on Poetry's task.

MarkKoz replied:
And you could also avoid useless stuff like collecting static files.

Edit: never mind that doesn't happen to begin with. I just got confused when viewing your old diff.

HassanAbouelela replied:
Added in 98fa159

HassanAbouelela committed:
Run Migrations For Tests

Call the migration handler in manage.py when running tests, before
handing control back to django.

Signed-off-by: Hassan Abouelela <hassan@hassanamr.com>

ChrisLovering committed:
Merge branch 'main' into speedup-tests
